libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -1085,40 +1085,31 @@ class MapboxRouteLineApi(
             routeProgress.currentLegProgress?.currentStepProgress,
             routePointsProvider(routeProgress.navigationRoute)
         ) { currentLegProgress, currentStepProgress, completeRoutePoints ->
-            var allRemainingPoints = 0
-
-            /**
-             * Finds the count of remaining points in the current step.
-             *
-             * TurfMisc.lineSliceAlong places an additional point at index 0 to mark the precise
-             * cut-off point which we can safely ignore.
-             * We'll add the distance from the upcoming point to the current's puck position later.
-             */
-            allRemainingPoints += try {
-                TurfMisc.lineSliceAlong(
-                    LineString.fromLngLats(currentStepProgress.stepPoints ?: emptyList()),
-                    currentStepProgress.distanceTraveled.toDouble(),
-                    currentStepProgress.step?.distance() ?: 0.0,
-                    TurfConstants.UNIT_METERS
-                ).coordinates().drop(1).size
-            } catch (e: TurfException) {
-                0
-            }
-
-            /**
-             * Add to the count of remaining points all of the remaining points on the current leg,
-             * after the current step.
-             */
+            val currentStepRemainingPoints: Int
+            val nextStepsRemainingPoints: Int
             if (currentLegProgress.legIndex < completeRoutePoints.stepPoints.size) {
                 val currentLegSteps = completeRoutePoints.stepPoints[currentLegProgress.legIndex]
-                allRemainingPoints += if (currentStepProgress.stepIndex < currentLegSteps.size) {
-                    currentLegSteps.slice(
+                if (currentStepProgress.stepIndex < currentLegSteps.size) {
+                    /**
+                     * Finds the count of remaining points in the current step.
+                     */
+                    currentStepRemainingPoints = currentLegSteps[currentStepProgress.stepIndex].size - currentStepProgress.geometryIndex - 1
```

Comment:
Why the `-1`? `currentStepProgress.geometryIndex` already marks the point _ahead of the user_, so for example:
```
[p u p p p]
```
where `p` is a point and `u` is user location, we should have a step geometry index of `1` (index of a point in front) and we should return 3 remaining points in the step, so:
```
currentStepRemainingPoints = currentLegSteps[currentStepProgress.stepIndex].size - currentStepProgress.geometryIndex
```
should be enough.

Reply:
I had it without -1 but it was breaking the tests...

> currentStepProgress.geometryIndex already marks the point ahead of the user

Are you sure? Then it can't be zero?

> where p is a point and u is user location, we should have a step geometry index of 1 (index of a point in front)

If you're saying that index should mark the point ahead of the user, index should be 2, isn't it? Because `u` has index 1 and the point ahead has index 2.

Reply:
Because the way stepGeometryIndex is calculated now is consistent with leg geometry index. See `NavigatorMapperStepGeometryIndexTest` so I'd say it's calculated correctly.

Reply:
> If you're saying that index should mark the point ahead of the user, index should be 2, isn't it? Because u has index 1 and the point ahead has index 2.

`u` (user) is not a point in the geometry of the step, I just added it for visualization purposes, the user is between points at index 0 and 1, so the next geometry index is 1.

> Are you sure? Then it can't be zero?

Right, it could only be zero for the first step if we haven't joined the route yet (basically the `INITIALIZED` state). That's my understanding.
cc @DenisPryt for visibility

We even document it this way in `RouteProgress`:
> index representing the geometry point right in front of the user

Reply:
I think the step geometry index is calculated correctly, something is off in the route line class, unless I'm missing something.

Reply:
Maybe it was wrong before with Turf?